lib/cli/src/Cardano/CLI.hs

```diff
@@ -866,14 +765,13 @@ cmdKeyPublic =
     cmd = fmap exec $
         KeyPublicArgs <$> keyArgument
 
-    exec (KeyPublicArgs key) = do
-        orReadStdin key >>= toPublic scheme >>= TIO.putStrLn
-
-    scheme :: CliKeyScheme Text IO
-    scheme =
-        hoistKeyScheme eitherToIO
-        $ defaultScheme Icarus
-
+    exec (KeyPublicArgs key) =
+        orReadStdin key
```

Comment:
This `orReadStdin key` feels a bit strange.

`key` is the key passed as argument. `orReadStdin key` reads the key from stdIn if the arguments are empty.

It's easy to forget it, or make the mistake of using `key` directly.

Maybe it can be moved to `cmd`.

Reply:
Or, simply make it the only way to provide a value. So, for key commands, inputs HAVE to be provided via stdin, and it's very easy to redirect "literals" if needed to make it like standard arguments:

```
echo ... | cardano-wallet key child
```